### ISSUE TITLE
Can't mount /Users on macos x using xhyve drive

### ISSUE THREAD

**Issue by cmoulliard** [minishift/minishift#1249]
Minishift version : 1.4.1
Driver : xhyve
MacOS Sierra : 10.12.5

We can't mount /Users on Macos X - using this command

```
minishift hostfolder add public
UNC path: //192.168.1.80/public
Mountpoint [/mnt/sda1/public]:
Username: dabou
Password: [HIDDEN]
Domain:
Added: public

minishift hostfolder mount public
   Mounting 'public': '//192.168.1.80/public' as '/mnt/sda1/public' ... FAIL
```

even if the folders is shared

![screenshot 2017-08-09 09 39 26](https://user-images.githubusercontent.com/463790/29110340-a76969a2-7ce6-11e7-9543-5ea54360e51e.png)

**Comment by gbraad** [comment]
is this using the B2D or CentOS image?

**Comment by cmoulliard** [comment]
I'm using B2D

**Comment by cmoulliard** [comment]
Here is the stack trace

```
minishift hostfolder mount users --show-libmachine-logs -v 5
Found binary path at /usr/local/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:60970
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) Calling .GetState
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mkdir -p /mnt/sda1/users
SSH cmd err, output: <nil>:
   Mounting 'users': '//192.168.1.80' as '/mnt/sda1/users' ... (minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
if grep -qs /mnt/sda1/users /proc/mounts; then echo '1'; else echo '0'; fi
SSH cmd err, output: <nil>: 0

(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo ping -c1 -w1 192.168.1.80
SSH cmd err, output: <nil>: PING 192.168.1.80 (192.168.1.80): 56 data bytes
64 bytes from 192.168.1.80: seq=0 ttl=64 time=0.194 ms

--- 192.168.1.80 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.194/0.194/0.194 ms

(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mkdir -p /mnt/sda1/users
SSH cmd err, output: <nil>:
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mount -t cifs //192.168.1.80 /mnt/sda1/users -o username=dabou,password=xxxx
SSH cmd err, output: exit status 255: mount: mounting //192.168.1.80 on /mnt/sda1/users failed

FAIL
Making call to close driver server
(minishift) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
```

**Comment by gbraad** [comment]
You need to specify the share. You can not mount the host. It needs something like //host/sharename to work

**Comment by gbraad** [comment]
Please specify \\host\shareable. I think the shell escapes the slash perhaps.

**Comment by cmoulliard** [comment]
Get a permission denied when I pass the shareable 

```
sudo mount -v -t cifs -o username=dabou,password=xxxx //192.168.1.80/Public /mnt/sda1/users
mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
mount error(13): Permission denied
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
mount: mounting //192.168.1.80/Public on /mnt/sda1/users failed: Permission denied
```

**Comment by cmoulliard** [comment]
I can mount it using sshfs at the condition that "Remote Login" is enable on the mach to start the ssh server

```
minishift ssh "sudo mkdir -p /Users/<username>"
minishift ssh "sudo chown -R docker /Users"
minishift ssh
sshfs -o sshfs_debug <username>@<IP_ADDRESS>:/Users/<username>/ /Users/<username> -o nonempty
```

Remark: I can't use the IP address defined for the bridge100 but instead the en4 to reach the macos machine within the boot2docker vm

**Comment by gbraad** [comment]
will try to reproduce it. might be an issue with the smb client on B2D.

**Comment by cmoulliard** [comment]
@gbraad Have you been able to reproduce ?

**Comment by cmoulliard** [comment]
> Please specify

Did it without success

```
minishift hostfolder add public
UNC path: \\192.168.1.80\public
Mountpoint [/mnt/sda1/public]: 
Username: dabou
Password: [HIDDEN]
Domain: 
Added: public

minishift hostfolder list         
Name     Mountpoint         Remote path             Mounted
public   /mnt/sda1/public   //192.168.1.80/public   N

minishift hostfolder mount public --show-libmachine-logs -v 5
Found binary path at /usr/local/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:52419
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) Calling .GetState
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mkdir -p /mnt/sda1/public
SSH cmd err, output: <nil>: 
   Mounting 'public': '//192.168.1.80/public' as '/mnt/sda1/public' ... (minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
if grep -qs /mnt/sda1/public /proc/mounts; then echo '1'; else echo '0'; fi
SSH cmd err, output: <nil>: 0

(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo ping -c1 -w1 192.168.1.80
SSH cmd err, output: <nil>: PING 192.168.1.80 (192.168.1.80): 56 data bytes
64 bytes from 192.168.1.80: seq=0 ttl=64 time=0.228 ms

--- 192.168.1.80 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.228/0.228/0.228 ms

(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mkdir -p /mnt/sda1/public
SSH cmd err, output: <nil>: 
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mount -t cifs //192.168.1.80/public /mnt/sda1/public -o username=dabou,password=xxx
SSH cmd err, output: exit status 32: mount error(115): Operation now in progress
Refer to the mount.cifs(8) manual page (e.g. man mount.cifs)
mount: mounting //192.168.1.80/public on /mnt/sda1/public failed: Operation now in progress

FAIL
Making call to close driver server
(minishift) Calling .Close
(minishift) DBG | Closing plugin on server side
Successfully made call to close driver server
Making call to close connection to plugin binary
```

**Comment by cmoulliard** [comment]
I get invalid argument within the busybock of boot2docker
when I try to use sec=ntlm

```
sudo mount -vvv -t cifs //192.168.64.1/public /mnt/sda1/public -o sec=ntlm,user=dabou,password=xxxx
```

**Comment by cmoulliard** [comment]
Finally, I was able to mount the shared folder between Macos and boot2docker busybox. The public folder which is shared by default on MacOS is not available under the name `/Public|/public` name but instead under `user's Public Folder`

This is why to resolve the issue, I have created another share for the Temp directory on Macos and that works

```
minishift hostfolder mount temp --show-libmachine-logs -v 5
Found binary path at /usr/local/bin/docker-machine-driver-xhyve
Launching plugin server for driver xhyve
Plugin server listening at address 127.0.0.1:54858
() DBG | operation not supported by device
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) Calling .GetState
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
sudo mkdir -p /mnt/sda1/temp
SSH cmd err, output: <nil>:
   Mounting 'temp': '//192.168.1.80/Temp' as '/mnt/sda1/temp' ... (minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/dabou/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.2 -o IdentitiesOnly=yes -i /Users/dabou/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
if grep -qs /mnt/sda1/temp /proc/mounts; then echo '1'; else echo '0'; fi
SSH cmd err, output: <nil>: 1

Already mounted
Making call to close driver server
(minishift) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
```

**Comment by hferentschik** [comment]
For what it's worth, I have been using a host folders on macOS for a long time. I have a persistent mount definition which I use for all instances. However, I use SMBUp to run an independent Samba server and configure my own shares.

**Comment by gbraad** [comment]
Please provide the configured shares information or the Samba config file

**Comment by hferentschik** [comment]
@cmoulliard Can we close this issue? Your issue is solved right?

**Comment by hmesha** [comment]
I'm getting the same issue. I'm getting the exact error. Would you please share the solution to the problem?

I'm running:
Minishift version 1.6
Driver : xhyve
MacOS Sierra : 10.12.5

**Comment by gbraad** [comment]
can you try SMBUp ?
macOS has been using several different versions of Samba over time now, and it is hard to keep up and test all of them.

**Comment by hmesha** [comment]
Thanks for your reply. 

I managed to get this working without switching my SMB to SMBUp. the trick was to enable the "Windows File Sharing" option on the share in OSX System Preferences. Here's how I did:
1. Go to System Preferences -> Sharing -> File Sharing. 
2. Select the share to be mounted by Minishift
3. Click options and enable "Windows File Sharing" for the user account.

Hopefully this can someone else save hours of troubleshooting.

-Hany
